### ISSUE TITLE
pickle doesn't work on mac m1 

### ISSUE THREAD

**Issue by 1k2s3v** [Doist/sqs-workers#16]
I've tried to use this library but had a problem with the pickled inside the library 

Hope this traceback helps you to fix this bug 

```
Traceback (most recent call last):
  File "/Users/svyatoslavkalina/Desktop/Projects/new/app.py", line 24, in <module>
    sqs.process_queues()
  File "/Users/svyatoslavkalina/Desktop/Projects/new/venv/lib/python3.10/site-packages/sqs_workers/sqs_env.py", line 132, in process_queues
    p.start()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/process.py", line 121, in start
    self._popen = self._Popen(self)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: cannot pickle 'module' object

Process finished with exit code 1
```

Example of code:
```py
# import pip
from sqs_workers import SQSEnv, create_standard_queue


sqs = SQSEnv()

create_standard_queue(sqs, "emails")
queue = sqs.queue("emails")


@queue.processor("send_email")
def send_email(to, subject, body):
    print('asd')

queue.add_job("send_email", to="shdevsir0@gmail.com", subject="Hello world", body="hello world")


sqs.process_queues()
```

**Comment by Schnouki** [comment]
Hey @1k2s3v, sorry for the delay.
Which version of sqs-workers are you using?

Could you please try to split your code in 2 files:

```py
# queue_handlers.py
from sqs_workers import SQSEnv, create_standard_queue

sqs = SQSEnv()

create_standard_queue(sqs, "emails")
queue = sqs.queue("emails")

@queue.processor("send_email")
def send_email(to, subject, body):
    print('asd')

queue.add_job("send_email", to="shdevsir0@gmail.com", subject="Hello world", body="hello world")
```

```py
# processor.py
from queue_handlers import sqs

def run():
    sqs.process_queues()

if __name__ == "__main__":
    run()
```

and run `python processor.py`?

**Comment by 1k2s3v** [comment]
Thank you for your response. But  I already figure out the problem and even forget how =)) Sory did not close this issue earlier. 
Have a nice day

**Comment by glsignal** [comment]
Hi there 🙂 I happened to have encountered the same problem as originally mentioned (`TypeError: cannot pickle 'module' object`)

I was wondering if you had any ways of helping to figure this out, whether it's an issue with my environment or implementation?

So far I've tried the minimal code mentioned in https://github.com/Doist/sqs-workers/issues/16#issuecomment-1285133877 and it still produces the problem. It's possible to avoid this with `multiprocessing.set_start_method("fork")` however that seems to present other issues affecting stability (at least for me, on this mac)

Calling something like this from the python repl (based on what the `process_queues` implementation does) will also produce the exception, I'm just at the limits of where to look further into detail as a relative newbie to python.

```python
>>> import pickle
>>> from queue_handlers import sqs
>>> fn = sqs.queue("email").process_queue
>>> fn
<bound method GenericQueue.process_queue of RawQueue(name='email', backoff_policy=<sqs_workers.backoff_policies.ExponentialBackoff object at 0x10d630710>, batching_policy=<sqs_workers.batching.NoBatching object at 0x10d5e6ad0>, processor=<function send_email at 0x10dcb91c0>)>
>>> pickle.dumps(fn)
...the computer screams into the void ...
TypeError: cannot pickle 'module' object
```

If you have any tips for investigating problems like this, I'd appreciate any pointers or direction!
Like the original reporter, I too hope you have (or are already having) a nice day 😄
